nodejsscan/path_traversal.yaml

```diff
@@ -14,8 +14,12 @@ rules:
         - pattern-inside: |
             $ELEC = require('electron');
             ...
-      - pattern-inside: |
-          $APP.$METHOD(..., function $FUNC($REQ, $RES, ...) { ... })
+      - pattern-either:
+        - pattern-inside: function ($REQ, $RES, ...) {...}
+        - pattern-inside: function $FUNC($REQ, $RES, ...) {...}
+        - pattern-inside: $X = function $FUNC($REQ, $RES, ...) {...}
+        - pattern-inside: var $X = function $FUNC($REQ, $RES, ...) {...};
```

Comment:
Do we have to explicitly mention `var`? The new versions of semgrep should already match it based on the rule at line 20?

Reply:
you absolutely right! there is no need to do explicitly mention `var` any more! :+1:

Reply:
...and I double checked, so for `pattern-inside` currently ` $X = function $FUNC($REQ, $RES, ...) {...}` will work only for `const` e.g
```javascript
const foo = function () { console.log(123) }
```
https://semgrep.live/AbDL

at the moment there are some differences between `pattern` and `pattern-inside`, there are issues for that in semgrep repo, but for this particular moment this is the most verbose way to cover express middleware/controllers

Reply:
Do you have an issue link in upstream semgrep that I can use to track this?

Reply:
well this one is slightly different https://github.com/returntocorp/semgrep/issues/1006
but I believe it refer to the same problem, I'll try to clarify it and open one more issue if needed

Reply:
Thanks subscribed to upstream